### PR TITLE
Method informative

### PR DESCRIPTION
**Pull request by lukyanoffpashok**

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#381](https://github.com/yegor256/aibolit/issues/381) is assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/dec7ddd1-5322-49eb-9623-4260fb82183f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: aibolit/model/model.py [hunk 1]
```diff
@@ -151,4 +148,72 @@ def predict(self, X, quantity_func='log', th=1.0):
...
+    def informative(self, X, return_acts=False):
+        """
+        Args:
+            X: np.array with shape (number of snippets, number of patterns) or
+                (number of patterns, ).
+        Returns:
+            ranked: np.array with shape (number of snippets, number of patterns)
+                of sorted patterns in non-increasing order for each snippet of
+                code.
+            acts: np.array with shape (number of snippets, ) of
```
Comment (lyriccoder) [inline aibolit/model/model.py:195]: explain pls what every number means:

1 - -do smth
2 - do different thing

Comment (lukyanoffpashok) [reply aibolit/model/model.py:195]: done

File: aibolit/model/model.py [hunk 2]
```diff
@@ -151,4 +148,73 @@ def predict(self, X, quantity_func='log', th=1.0):
...
+
+    def get_array(self, X, mask, i, incr):
+        """
+        Args:
+            X: np.array with shape (number of snippets, number of patterns).
+            mask: np.array with shape (number of snippets, number of patterns).
+            i: int, 0 <= i < number of patterns.
+            add: bool.
+        Returns:
+            X1: modified np.array with shape (number of snippets, ).
```
Comment (acheshkov) [inline aibolit/model/model.py:163]: are you sure that shape is (number of snippets, ) ?

File: aibolit/model/model.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def sigmoid(x):
     def __vstack_arrays(self, res):
         return np.vstack(res).T
 
-    def predict(self, X, quantity_func='log', th=1.0):
+    def predict(self, X, return_acts=False, quantity_func='log', th=1.0):
```
Comment (acheshkov) [inline aibolit/model/model.py:115]: what is the return_acts parameter?

File: aibolit/model/model.py [hunk 4]
```diff
@@ -151,4 +148,73 @@ def predict(self, X, quantity_func='log', th=1.0):
...
+            elemental minimum of 3 arrays.
+            number: np.array with shape (number of snippets, ) of
+            arrays' numbers with minimum elements.            .
+        """
+
+        c = np.vstack((c1, c2, c3))
+
+        return np.min(c, 0), np.argmin(c, 0)
+
+    def informative(self, X, return_acts=False):
```
Comment (acheshkov) [inline aibolit/model/model.py:186]: what is the return_acts parameter?

Comment (lukyanoffpashok) [reply aibolit/model/model.py:186]: If true, return acts, needed to do with most important pattern (np changes, -1, +1)

File: aibolit/model/model.py (hunk 2 above)
Comment (lukyanoffpashok) [reply aibolit/model/model.py]: fix

File: aibolit/model/model.py (hunk 3 above)
Comment (lukyanoffpashok) [reply aibolit/model/model.py:115]: check method informative

**Comment by lukyanoffpashok** [comment]
Retry DeepCode

**Comment by lukyanoffpashok** [comment]
Retry DeepCode

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job audit: performer @lyriccoder[/z](https://www.0crat.com/u/lyriccoder)[/z](https://www.0crat.com/u/lyriccoder) didn't make CR comments

**Pull request merged**